compiler/rustc_middle/src/mir/syntax.rs

```diff
@@ -180,6 +180,53 @@ pub enum BorrowKind {
     Mut { kind: MutBorrowKind },
 }
 
+#[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, TyEncodable, TyDecodable)]
+#[derive(Hash, HashStable)]
+pub enum RawPtrKind {
+    Mut,
+    Not,
```

Comment:
"Not" is an odd choice here. It's not a kind...?

Should probably be `Const`, since it corresponds to `&raw const`.

Reply:
`Mutability::Not` makes sense, but `RawPtrKind::Not` doesn't really.